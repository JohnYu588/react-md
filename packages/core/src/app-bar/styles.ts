import { cnb } from "cnbuilder";
import { type BoxOptions, box } from "../box/styles.js";
import { type BackgroundColor, cssUtils } from "../cssUtils.js";
import { type CssPosition } from "../types.js";
import { bem } from "../utils/bem.js";

declare module "react" {
  interface CSSProperties {
    "--rmd-app-bar-height"?: string | number;
    "--rmd-app-bar-background-color"?: string;
    "--rmd-app-bar-color"?: string;
    "--rmd-app-bar-surface-background-color"?: string;
    "--rmd-app-bar-surface-color"?: string;
  }
}

const styles = bem("rmd-app-bar");
const titleStyles = bem("rmd-app-bar-title");

export type AppBarPosition = "top" | "bottom";

/**
 * - `"clear"` - the background color will be transparent
 * - `"primary"` - defaults to the current primary theme color
 * - `"secondary"` - defaults to the current secondary theme color
 * - `"surface"` - this will use the current surface background color which
 *   defaults to `#fff` for light themes, `#242424` for dark themes, and
 *   `#424242` for dark themes when the `$disable-dark-elevation` is set to
 *   `true`
 *
 * The default dark theme surface color also depends on the `$fixed-elevation`
 * value.
 */
export type AppBarTheme = BackgroundColor | "clear";

/**
 * - `"auto"` - the height will be determined by the height of the content
 * - `"normal"` - defaults to `3.5rem` (`$height`)
 * - `"prominent"` - defaults to `7rem` (`$prominent-height`)
 * - `"dense"` - defaults to `3rem` (`$dense-height`)
 * - `"prominent-dense"` - defaults to `6rem` (`$prominent-dense-height`)
 */
export type AppBarHeight =
  | "auto"
  | "normal"
  | "prominent"
  | "dense"
  | "prominent-dense";

/** @since 6.0.0 */
export interface AppBarClassNameOptions extends Omit<BoxOptions, "fullWidth"> {
  className?: string;

  /**
   * Set this to `"fixed"` or `"sticky"` to set `position: fixed;` or
   * `position: sticky;` to the app bar. The default position will be static and
   * inline with other content.
   *
   * @defaultValue `'static'`
   */
  position?: CssPosition;

  /**
   * The position within the page to "fix" the `AppBar` when the `fixed` prop is
   * enabled.
   *
   * @defaultValue `"top"`
   */
  pagePosition?: AppBarPosition;

  /**
   * Set this to `true` to remove the box-shadow.
   *
   * @defaultValue `false`
   */
  disableElevation?: boolean;

  /**
   * The theme to apply to the `AppBar`.
   *
   * @defaultValue `"primary"`
   * @see {@link AppBarTheme}
   */
  theme?: AppBarTheme;

  /**
   * @defaultValue `stacked ? "auto" : "normal"`
   * @see {@link AppBarHeight}
   */
  height?: AppBarHeight;

  /** @see {@link CssUtilsOptions.surfaceColor} */
  surfaceColor?: "light" | "dark";

  /**
   * Set this to `true` if the app bar's positioning and width should be
   * changed whenever the scrollbar is visible on the page. This defaults to
   * `true` when the {@link position} prop is `true` so that once dialogs and menus
   * become visible the contents in the app bar do not need to be repainted.
   *
   * @since 6.0.0
   * @defaultValue `position === "fixed"`
   */
  scrollbarOffset?: boolean;
}

/**
 * Apply the `className`s for a tree component. This will be type-safe if using
 * typescript.
 *
 * @since 6.0.0
 */
export function appBar(options: AppBarClassNameOptions = {}): string {
  const {
    className,
    theme = "primary",
    stacked,
    height = stacked ? "auto" : "normal",
    position = "static",
    pagePosition = "top",
    scrollbarOffset = position === "fixed",
    align,
    grid,
    gridColumns,
    gridName,
    justify,
    reversed,
    surfaceColor,
    disableWrap = true,
    disablePadding,
    disableElevation,
  } = options;
  const surface = theme === "surface";
  const clear = theme === "clear";

  return cnb(
    styles({
      surface,
      [height]: height !== "normal",
      fixed: position !== "static",
      stacked,
      sticky: position === "sticky",
      [pagePosition]: position !== "static",
      elevated: position !== "static" && !disableElevation,
      "scrollbar-offset": scrollbarOffset,
      "static-scrollbar-offset": position === "static" && scrollbarOffset,
    }),
    box({
      align,
      stacked,
      fullWidth: true,
      grid,
      gridColumns,
      gridName,
      justify,
      reversed,
      disableWrap,
      disablePadding,
    }),
    cssUtils({
      backgroundColor: !surface && !clear ? theme : undefined,
      surfaceColor,
    }),
    className
  );
}

/**
 * - `"small"` - the first character in the title will be `1rem` (`$keyline`)
 *   from the edge of the app bar horizontally
 * - `"nav"` - this should be set when there is a nav button before the title so
 *   that the first character in the title will be `4.5rem` (`title-keyline`)
 * - `"list"` - this should be used when the title should align with the list
 *   item keyline and there is no nav icon before.
 *
 * @since 6.0.0
 */
export type AppBarTitleKeyline = "small" | "nav" | "list";

/** @since 6.0.0 */
export interface AppBarTitleClassNameOptions {
  className?: string;

  /**
   * @defaultValue `"small"`
   * @see {@link AppBarTitleKeyline}
   */
  keyline?: AppBarTitleKeyline;
}

/**
 * Apply the `className`s for a `AppBarTitle` component. This will be type-safe
 * if using typescript.
 *
 * @since 6.0.0
 */
export function appBarTitle(options: AppBarTitleClassNameOptions = {}): string {
  const { className, keyline = "small" } = options;
  return cnb(
    titleStyles({
      keyline: keyline == "list",
      "nav-keyline": keyline === "nav",
    }),
    className
  );
}
