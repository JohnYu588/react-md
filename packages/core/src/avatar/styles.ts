import { cnb } from "cnbuilder";
import { bem } from "../utils/bem";

const styles = bem("rmd-avatar");

/** @remarks \@since 6.0.0 */
export interface AvatarClassNameOptions {
  className?: string;

  /** @defaultValue `""` */
  color?: string;

  /** @defaultValue `"avatar"` */
  size?: "avatar" | "icon";
}

/**
 * @remarks \@since 6.0.0
 */
export function avatar(options: AvatarClassNameOptions): string {
  const { className, color = "", size } = options;

  return cnb(styles({ [color]: color, icon: size === "icon" }), className);
}

/** @remarks \@since 6.0.0 */
export interface AvatarImageClassNameOptions {
  className?: string;
}

/**
 * @remarks \@since 6.0.0
 */
export function avatarImage(options: AvatarImageClassNameOptions = {}): string {
  const { className } = options;

  return cnb(styles("image"), className);
}
