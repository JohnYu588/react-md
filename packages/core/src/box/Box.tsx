import { type HTMLAttributes, forwardRef } from "react";

import { type BoxOptions, box } from "./styles.js";

/**
 * @since 6.0.0
 */
export interface BoxProps extends HTMLAttributes<HTMLDivElement>, BoxOptions {}

/**
 * The `Box` component is a wrapper around the CSS box model and should solve
 * most of your `flex` and `grid` layout requirements for responsive design.
 * There are pass-through props for all of the box module styling properties
 * available by default.
 *
 * @example Default Styles
 * ```scss
 * .box {
 *   align-items: center;
 *   display: flex;
 *   flex-wrap: wrap;
 *   gap: 1rem;
 *   padding: 1rem;
 * }
 * ```
 *
 * ```tsx
 * import { Box } from "@react-md/core";
 * import type { ReactElement } from "react";
 *
 * export default function Example(): ReactElement {
 *   return (
 *     <Box>
 *       <div>Thing 1</div>
 *       <div>Thing 2</div>
 *       <div>Thing 3</div>
 *       <div>Thing 4</div>
 *       <div>Thing 5</div>
 *     </Box>
 *   ):
 * }
 * ```
 *
 * @example Default Grid Styles
 * ```scss
 * .box {
 *   align-items: center;
 *   display: grid;
 *   grid-gap: 1rem;
 *   grid-template-columns: repeat(auto-fit, minmax(8rem, 1fr));
 *   padding: 1rem;
 * }
 * ```
 *
 * ```tsx
 * import { Box } from "@react-md/core";
 * import type { ReactElement } from "react";
 *
 * export default function Example(): ReactElement {
 *   return (
 *     <Box grid>
 *       <div>Thing 1</div>
 *       <div>Thing 2</div>
 *       <div>Thing 3</div>
 *       <div>Thing 4</div>
 *       <div>Thing 5</div>
 *     </Box>
 *   ):
 * }
 * ```
 *
 * @example Custom Grid
 * ```scss
 * @use "@react-md/core" with (
 *   $box-grids: (
 *     small: (
 *       min: 5rem,
 *     ),
 *     medium: (
 *       min: 7rem,
 *       padding: 2rem,
 *       gap: 0.5rem,
 *     ),
 *   )
 * );
 *
 * ```
 *
 * ```tsx
 * import { Box } from "@react-md/core";
 * import type { ReactElement } from "react";
 *
 * export default function Example(): ReactElement {
 *   return (
 *     <Box grid gridName="medium">
 *       <div>Thing 1</div>
 *       <div>Thing 2</div>
 *       <div>Thing 3</div>
 *       <div>Thing 4</div>
 *       <div>Thing 5</div>
 *     </Box>
 *   ):
 * }
 * ```
 *
 * @since 6.0.0
 */
export const Box = forwardRef<HTMLDivElement, BoxProps>(
  function Box(props, ref) {
    const {
      style: propStyle,
      className,
      children,
      align,
      grid,
      gridName,
      justify,
      stacked,
      reversed,
      gridColumns,
      fullWidth,
      disableGap,
      disableWrap,
      disablePadding,
      ...remaining
    } = props;

    let style = propStyle;
    if (typeof gridColumns === "number") {
      style = {
        ...propStyle,
        "--rmd-box-columns": gridColumns,
      };
    }

    return (
      <div
        {...remaining}
        style={style}
        ref={ref}
        className={box({
          className,
          fullWidth,
          grid,
          stacked,
          gridName,
          gridColumns,
          align,
          justify,
          reversed,
          disableGap,
          disableWrap,
          disablePadding,
        })}
      >
        {children}
      </div>
    );
  }
);
