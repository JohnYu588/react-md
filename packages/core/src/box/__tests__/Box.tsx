import { describe, expect, it } from "@jest/globals";
import { render, screen, userEvent } from "../../test-utils/index.js";
import { Box, type BoxProps } from "../Box.js";
import { createRef } from "react";
import { type BoxJustifyContent, type BoxAlignItems } from "../styles.js";

const align: BoxAlignItems[] = [
  "start",
  "flex-start",
  "center",
  "end",
  "flex-end",
  "stretch",
];

const justify: BoxJustifyContent[] = [
  ...align,
  "space-between",
  "space-around",
  "space-evenly",
];

const BASE_PROPS: BoxProps = {
  "data-testid": "box",
  children: "Content",
};

describe("Box", () => {
  it("should apply the correct styling, HTML attributes, and allow a ref", () => {
    const ref = createRef<HTMLDivElement>();
    const props = {
      ...BASE_PROPS,
      ref,
    } as const;
    const { rerender } = render(<Box {...props} />);

    const box = screen.getByTestId("box");
    expect(ref.current).toBeInstanceOf(HTMLDivElement);
    expect(ref.current).toBe(box);
    expect(box).toMatchSnapshot();

    rerender(
      <Box
        {...props}
        style={{ color: "white" }}
        className="custom-class-name"
      />
    );
    expect(box).toMatchSnapshot();
  });

  it("should support common flex behavior", () => {
    const { rerender } = render(<Box {...BASE_PROPS} />);

    const box = screen.getByTestId("box");
    align.forEach((align) => {
      justify.forEach((justify) => {
        rerender(<Box {...BASE_PROPS} align={align} justify={justify} />);

        expect(box).toMatchSnapshot();
      });
    });
  });

  it("should allow the padding to be disabled", () => {
    const { rerender } = render(<Box {...BASE_PROPS} />);
    const box = screen.getByTestId("box");

    expect(box).toHaveClass("rmd-box--padded");

    rerender(<Box {...BASE_PROPS} disablePadding />);
    expect(box).not.toHaveClass("rmd-box--padded");
  });

  it("should support a full width prop to expand within other flex/grid containers", () => {
    const { rerender } = render(<Box {...BASE_PROPS} />);
    const box = screen.getByTestId("box");

    expect(box).not.toHaveClass("rmd-box--full-width");

    rerender(<Box {...BASE_PROPS} fullWidth />);
    expect(box).toHaveClass("rmd-box--full-width");
  });

  it("should be able to render as a grid using the grid prop", () => {
    render(<Box {...BASE_PROPS} grid />);
    const box = screen.getByTestId("box");
    expect(box).toMatchSnapshot();
  });

  it("should support rendering the grid with auto-fill instead of auto-fit", () => {
    const { rerender } = render(<Box {...BASE_PROPS} grid />);
    const box = screen.getByTestId("box");

    expect(box).toHaveClass("rmd-box--grid");
    expect(box).not.toHaveClass("rmd-box--grid-fill");

    rerender(<Box {...BASE_PROPS} grid gridColumns="fit" />);
    expect(box).toHaveClass("rmd-box--grid");
    expect(box).not.toHaveClass("rmd-box--grid-fill");

    rerender(<Box {...BASE_PROPS} grid gridColumns="fill" />);
    expect(box).toHaveClass("rmd-box--grid");
    expect(box).toHaveClass("rmd-box--grid-fill");
  });

  it("should set the --rmd-box-columns variable and grid-columsn class when gridColumns is a number", () => {
    const { rerender } = render(<Box {...BASE_PROPS} grid />);
    const box = screen.getByTestId("box");

    expect(box).toMatchSnapshot();
    expect(box).toHaveClass("rmd-box--grid");
    expect(box).not.toHaveClass("rmd-box--grid-fill");
    expect(box).not.toHaveClass("rmd-box--grid-columns");

    rerender(<Box {...BASE_PROPS} grid gridColumns={4} />);
    expect(box).toMatchSnapshot();
    expect(box).toHaveClass("rmd-box--grid");
    expect(box).not.toHaveClass("rmd-box--grid-fill");
    expect(box).toHaveClass("rmd-box--grid-columns");

    rerender(
      <Box {...BASE_PROPS} grid gridColumns={4} style={{ color: "red" }} />
    );
    expect(box).toMatchSnapshot();
  });

  it("should be able to generate a grid name", () => {
    render(<Box {...BASE_PROPS} grid gridName="variant-1" />);
    const box = screen.getByTestId("box");

    expect(box).toHaveClass("rmd-box--variant-1");
  });

  it("should be able stack and reverse ordering", () => {
    const { rerender } = render(<Box {...BASE_PROPS} />);
    const box = screen.getByTestId("box");

    expect(box).not.toHaveClass("rmd-box--column");
    expect(box).not.toHaveClass("rmd-box--reverse");
    expect(box).not.toHaveClass("rmd-box--column-reverse");

    rerender(<Box {...BASE_PROPS} stacked />);
    expect(box).toHaveClass("rmd-box--column");
    expect(box).not.toHaveClass("rmd-box--reverse");
    expect(box).not.toHaveClass("rmd-box--column-reverse");

    rerender(<Box {...BASE_PROPS} stacked reversed />);
    expect(box).not.toHaveClass("rmd-box--column");
    expect(box).not.toHaveClass("rmd-box--reverse");
    expect(box).toHaveClass("rmd-box--column-reverse");

    rerender(<Box {...BASE_PROPS} reversed />);
    expect(box).not.toHaveClass("rmd-box--column");
    expect(box).toHaveClass("rmd-box--reverse");
    expect(box).not.toHaveClass("rmd-box--column-reverse");
  });
});
