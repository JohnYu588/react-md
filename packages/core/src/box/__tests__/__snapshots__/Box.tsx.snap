// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should allow the columns to be configured by different breakpoints 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-phone"
  data-testid="box"
  style=""
>
  Content
</div>
`;

exports[`Box should allow the columns to be configured by different breakpoints 2`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-phone rmd-box--grid-phone-size rmd-box--grid-tablet rmd-box--grid-desktop"
  data-testid="box"
  style="--rmd-box-phone-columns: 1;"
>
  Content
</div>
`;

exports[`Box should allow the columns to be configured by different breakpoints 3`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-tablet rmd-box--grid-tablet-size rmd-box--grid-desktop rmd-box--grid-desktop-size rmd-box--grid-large-desktop"
  data-testid="box"
  style="--rmd-box-tablet-columns: 2; --rmd-box-desktop-columns: 4;"
>
  Content
</div>
`;

exports[`Box should allow the grid item size to be configured by different breakpoints 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid"
  data-testid="box"
  style="--rmd-box-item-min-size: 4rem;"
>
  Content
</div>
`;

exports[`Box should allow the grid item size to be configured by different breakpoints 2`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-phone rmd-box--grid-phone-size"
  data-testid="box"
  style="--rmd-box-phone-item-min-size: 4rem;"
>
  Content
</div>
`;

exports[`Box should allow the grid item size to be configured by different breakpoints 3`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-phone rmd-box--grid-phone-size rmd-box--grid-tablet rmd-box--grid-tablet-size rmd-box--grid-desktop rmd-box--grid-desktop-size"
  data-testid="box"
  style="--rmd-box-phone-item-min-size: 4rem; --rmd-box-tablet-item-min-size: 8rem; --rmd-box-desktop-item-min-size: 12rem;"
>
  Content
</div>
`;

exports[`Box should allow the grid item size to be configured by different breakpoints 4`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-tablet rmd-box--grid-tablet-size rmd-box--grid-desktop rmd-box--grid-desktop-size rmd-box--grid-large-desktop rmd-box--grid-large-desktop-size"
  data-testid="box"
  style="--rmd-box-tablet-item-min-size: 5rem; --rmd-box-desktop-item-min-size: 12rem; --rmd-box-large-desktop-item-min-size: auto;"
>
  Content
</div>
`;

exports[`Box should apply the correct styling, HTML attributes, and allow a ref 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should apply the correct styling, HTML attributes, and allow a ref 2`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded custom-class-name"
  data-testid="box"
  style="color: white;"
>
  Content
</div>
`;

exports[`Box should be able to render as a grid using the grid prop 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should be able to support the grid-auto-rows behavior 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should be able to support the grid-auto-rows behavior 2`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should be able to support the grid-auto-rows behavior 3`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-auto-rows"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should set the --rmd-box-columns variable and grid-columsn class when gridColumns is a number 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should set the --rmd-box-columns variable and grid-columsn class when gridColumns is a number 2`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-size"
  data-testid="box"
  style="--rmd-box-columns: 4;"
>
  Content
</div>
`;

exports[`Box should set the --rmd-box-columns variable and grid-columsn class when gridColumns is a number 3`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--grid rmd-box--grid-size"
  data-testid="box"
  style="--rmd-box-columns: 4; color: red;"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 1`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 2`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 3`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-center"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 4`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 5`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 6`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-stretch"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 7`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-between"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 8`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-around"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 9`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-evenly"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 10`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 11`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 12`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-center"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 13`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 14`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 15`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-stretch"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 16`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-between"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 17`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-around"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 18`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-start rmd-box--justify-evenly"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 19`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 20`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 21`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-center"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 22`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 23`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 24`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-stretch"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 25`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-between"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 26`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-around"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 27`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-center rmd-box--justify-evenly"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 28`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 29`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 30`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-center"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 31`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 32`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 33`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-stretch"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 34`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-between"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 35`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-around"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 36`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-evenly"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 37`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 38`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 39`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-center"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 40`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 41`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 42`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-stretch"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 43`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-between"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 44`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-around"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 45`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-end rmd-box--justify-evenly"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 46`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 47`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-start"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 48`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-center"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 49`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 50`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-end"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 51`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-stretch"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 52`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-between"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 53`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-around"
  data-testid="box"
>
  Content
</div>
`;

exports[`Box should support common flex behavior 54`] = `
<div
  class="rmd-box rmd-box--gap rmd-box--wrap rmd-box--padded rmd-box--align-stretch rmd-box--justify-evenly"
  data-testid="box"
>
  Content
</div>
`;
